common-test/src/main/java/io/strimzi/test/k8s/BaseKubeClient.java

```diff
@@ -234,8 +234,20 @@ public K waitForDeployment(String name) {
 
     @Override
     public K waitForPod(String name) {
+        // wait when all pods are ready
         return waitFor("pod", name,
-            actualObj -> "Running".equals(actualObj.get("status").get("phase").asText()));
+            actualObj -> {
+                JsonNode containerStatuses = actualObj.get("status").get("containerStatuses");
+                ArrayList<Boolean> ready = new ArrayList<>();
+                if (containerStatuses.isArray()) {
+                    for (final JsonNode objNode : containerStatuses) {
+                        ready.add(objNode.get("ready").asBoolean());
```

Comment:
No need for `ready`: If `if (!objNode.get("ready").asBoolean()) return false` and if you get all the way through the loop then `return true`.